src/test/java/uk/gov/hmcts/reform/migration/repository/ElasticSearchRepositoryTest.java

```diff
@@ -0,0 +1,134 @@
+package uk.gov.hmcts.reform.migration.repository;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.ccd.client.model.SearchResult;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ElasticSearchRepositoryTest {
+
+    private static final String USER_TOKEN = "TEST_USER_TOKEN";
+
+    private static final String CASE_TYPE = "CASE_TYPE";
+
+    private static final String AUTH_TOKEN = "Test_Auth_Token";
+
+    private static final String INITIAL_QUERY = "{\n"
+        + "  \"query\": {\n"
+        + "    \"match_all\": {}\n"
+        + "  },\n"
+        + "  \"_source\": [\n"
+        + "    \"reference\"\n"
+        + "  ],\n"
+        + "  \"size\": 100,\n"
+        + "  \"sort\": [\n"
+        + "    {\n"
+        + "      \"reference.keyword\": \"asc\"\n"
+        + "    }\n"
+        + "  ]\n"
+        + "\n"
+        + "}";
+
+    private static final String SEARCH_AFTER_QUERY = "{\n"
+        + "  \"query\": {\n"
+        + "    \"match_all\": {}\n"
+        + "  },\n"
+        + "  \"_source\": [\n"
+        + "    \"reference\"\n"
+        + "  ],\n"
+        + "  \"size\": 100,\n"
+        + "  \"sort\": [\n"
+        + "    {\n"
+        + "      \"reference.keyword\": \"asc\"\n"
+        + "    }\n"
+        + "  ]\n"
+        + ",\"search_after\": [1677777777]\n"
+        + "}";
+
+    private static final int QUERY_SIZE = 100;
+
+    private ElasticSearchRepository elasticSearchRepository;
+
+    @Mock
+    private CoreCaseDataApi coreCaseDataApi;
+
+    @Mock
+    private AuthTokenGenerator authTokenGenerator;
+
+    @Before
+    public void setUp() {
+        elasticSearchRepository = new ElasticSearchRepository(coreCaseDataApi, authTokenGenerator, QUERY_SIZE);
+        when(authTokenGenerator.generate()).thenReturn(AUTH_TOKEN);
+    }
+
+    @Test
+    public void shouldReturnSearchResultsForCaseTypeElasticSearch() {
```

Comment:
@kiran-yenigala-hmcts we can add null (for both searchResult and subsequentSearchResult) and empty (searchResult) check test cases too

Reply:
addressed